### ISSUE TITLE
Horizon does not release a failed job and fail all the subsequent jobs

### ISSUE THREAD

**Issue by shoosah** [laravel/horizon#1248]
- Horizon Version: 5.10.6
- Laravel Version: 9.28.0
- PHP Version: 8.1
- Redis Driver & Version: predis/phpredis 2.0.3
- Database Driver & Version: MySQL 5.7	

### Description:
We've been using Laravel Queue with the following settings:
```
'queue' => [
            'driver'      => env('QUEUE_DRIVER', 'redis'),
            'connection'  => 'queue',
            'retry_after' => 10,
            'queue'       => 'default',
        ],
```

Please note the following environment variables:
```
QUEUE_DRIVER=redis
QUEUE_CONNECTION=queue
QUEUE_FAILED_DRIVER=database
```

And if a job fails because of an unknown error, it throws an error in the system__queue_failures table as follows:
```
Illuminate\Queue\MaxAttemptsExceededException: App\Jobs\ProcessPayment has been attempted too many times or run too long. The job may have previously timed out. in /container/application/versions/.../project/vendor/laravel/framework/src/Illuminate/Queue/Worker.php:750
```

Since started using Horizon, when the same unknown error happens, the job is marked as failed. However, it is not logged in the DB and also all the subsequent jobs fail.

### Steps To Reproduce:
ProcessPayment job should be responsible for different types of payment classes which pass jobs to it. For example class paymentA and class paymentB. 
- Throw an exception in class paymentA. 
- After that start Horizon using `php artisan horizon`.
- Then throw a job at ProcessPayment queue using paymentA class. This should fail and create a new record in `system__queue_failures` table.  The job fails but no record is being created in `system__queue_failures` table.
- After that, try to use paymentB (which doesn't throw any exception) which adds another job in ProcessPayment. However, the job immediately fails until we stop running `php artisan horizon` and start again.

```
Horizon started successfully.
   INFO  Processing jobs from the [payments] queue.  

  2023-02-16 00:58:40 App\Jobs\ProcessPayment ................ 6,858.04ms DONE
  2023-02-16 00:59:02 App\Jobs\ProcessPayment ................ 3,358.03ms FAIL
  2023-02-16 00:59:57 App\Jobs\ProcessPayment ................... 14.78ms FAIL
```

**Comment by ankurk91** [comment]
Might be related 
https://github.com/laravel/framework/discussions/43502

**Comment by shoosah** [comment]
I suggest to close this issue. I figured the same issue was happening when I was running `php artisan queue:work queue --queue=payments`. What I didn't realise that I set `--max-jobs=1`. Therefore, this issue is related to the way that Laravel queue handles jobs. What wasn't very obvious in Horizon doc is that I could've set `maxJobs => 1` to replicate the same behaviour as `php artisan queue:work queue --queue=payments --max-jobs=1`
